MIP22/mip22.md

```diff
@@ -140,41 +262,45 @@ The `Tinlake Flipper` takes all of the collateral from a Vault and requests rede
     }
 ```
 
-The redeem order will now be taken into account whenever loan repayments flow into the pool or other investors want to join the pool. As the pool redeems the DROP any user can call the function `take`. It moves the DAI available for withdrawal from the pool and either repays up outstanding debt (`tab`) to the `vow` or returns it to the pool. This method can be called multiple time to cause the pool to `disburse` more DAI and return it to the pool as funds available for borrowers.
+As DAI is redeemed from the Pool, the manager transfers the DAI into the `vow` until the `tab` is recovered. At that point any additional DAI that could be redeemed with the DROP is given back to the Pool.
+
+The redeem order will now be taken into account whenever loan repayments flow into the pool or other investors want to join the pool. As the pool redeems the DROP any user can call the function `recover`. It moves the DAI available for withdrawal from the pool and either repays up outstanding debt (`tab`) to the `vow` or returns it to the pool. This method can be called multiple time to cause the pool to `disburse` more DAI and return it to the pool as funds available for borrowers.
 
 ```
-    function take() public {
-        (uint returned, ) = pool.disburse();
-        uint tabWad = tab / RAY; // always rounds down, this could lead to < 1 RAY to be lost in dust
+    function recover(uint endEpoch) public {
+        require(!safe && !glad && live, "TinlakeManager/Pool-healhty");
+        (uint returned, , ,) = pool.disburse(endEpoch);
+
+        uint tabWad = tab / ONE; // always rounds down, this could lead to < 1 RAY to be lost in dust
         if (tabWad < returned) {
-            dai.transfer(dest, sub(returned, tabWad));
+            dai.transfer(owner, sub(returned, tabWad));
             returned = tabWad;
         }
         if (tab != 0) {
             daiJoin.join(vow, returned);
-            tab = sub(tab, mul(returned, RAY));
+            tab = sub(tab, mul(returned, ONE));
         }
+
     }
 ```
 
-The Tinlake smart contracts collect redeem orders and return any DAI returned by borrowers proportionally to all DROP token holders first. This means that to receive highest priority on the redemption the liquidation should request the entire DROP to be redeemed.
 
-This can be done by setting the variable `dunk` to `uint(-1)`. This raises the minimum size of the collateral to auction off during the liquidation to above the maximum collateral that can ever exist thus forcing liquidation of the entire Vault and not just part of it.
+To ensure that the `Cat` always `bites` the entire vault, the `dunk` of the collateral type should be set to MAX_UINT. This raises the minimum size of the collateral to auction off during the liquidation to above the maximum collateral that can ever exist thus forcing liquidation of the entire Vault and not just part of it.
```

Comment:
I'm going to flag this `dunk` idea as possibly dangerous.  We found that `dunk` can overflow causing a revert and in turn prevent liquidations from happening.  Here is a comment that mentions `dunk` overflowing at roughly `100 trillion DAI`:

https://github.com/makerdao/dss/blob/master/src/cat.sol#L163-L165

I would not assume that comment is correct, and would set `dunk` to perhaps 99 trillion and test to make sure we can still liquidated at these insane magnitudes.  For now, we might just want to say we are limiting `dunk` to `99 trillion DAI`, but test for it in the deploy tests.  In fact, our integration tests test to make sure this is below a million:

https://github.com/makerdao/spells-mainnet/blob/master/src/DssSpell.t.sol#L615

Reply:
I just clarified this detail and our spell implementation will set it to a multiple of the debt ceiling.

Reply:
Looks good.